coconut/configuration/configurationutil.go

```diff
@@ -0,0 +1,224 @@
+			components = append(components, key+"@"+value)
+		} else {
+			components = append(components, key)
+		}
+	}
+	return components, nil, nonZero
+}
+
+func drawTableHistoryConfigs(headers []string, history []string, max int, o io.Writer) {
+	table := tablewriter.NewWriter(o)
+	if len(headers) > 0 {
+		table.SetHeader(headers)
+	}
+	table.SetBorder(false)
+	table.SetColMinWidth(0, max)
+
+	for _, value := range history {
+		component, entry, timestamp := getComponentEntryTimestampFromConsul(value)
+		prettyTimestamp, err := getTimestampInFormat(timestamp, time.RFC822)
+		if err != nil {
+			prettyTimestamp = timestamp
+		}
+		configName := red(component) + "/" + blue(entry) + "@" + timestamp
+		table.Append([]string{configName, prettyTimestamp})
+	}
+	table.Render()
+}
+
+func formatListOutput( cmd *cobra.Command, output []string)(parsedOutput []byte, err error) {
+	format, err := cmd.Flags().GetString("output")
+	if err != nil {
+		return
+	}
+
+	switch strings.ToLower(format) {
+	case "json":
+		parsedOutput, err = json.MarshalIndent(output, "", "    ")
+	case "yaml":
+		parsedOutput, err = yaml.Marshal(output)
+	}
+	if err != nil {
+		return
+	}
+	return parsedOutput, nil
+}
+
+// Method to split component, entry and timestamp when being passed a key from consul
+// e.g. of key o2/components/quality-control/cru-demo/12345678
+func getComponentEntryTimestampFromConsul(key string)(string, string, string) {
+	key = strings.TrimPrefix(key, componentsPath)
+	key = strings.TrimPrefix(key, "/'")
+	key = strings.TrimSuffix(key, "/")
+	elements := strings.Split(key, "/")
+	return elements[0], elements[1], elements[2]
+}
+
+func getMaxLenOfKey(keys []string) (maxLen int){
+	maxLen = 0
+	for _, value := range keys {
+		if len(value) - len(componentsPath) >= maxLen {
+			maxLen = len(value) - len(componentsPath)
+		}
+	}
+	return
+}
+
+func getFileContent(filePath string)(fileContent []byte, err error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return
+	}
+	defer file.Close()
+
+	fileContentByte, err := ioutil.ReadAll(file)
+	if err != nil {
+		return
+	}
+	return fileContentByte, nil
+}
+
+func getComponentsMapFromKeysList(keys []string) map[string]bool {
+	var componentsMap = make(map[string]bool)
+	for _,value := range keys {
+		value := strings.TrimPrefix(value, componentsPath)
+		component := strings.Split(value, "/" )[0]
+		componentsMap[component] = true
+	}
+	return componentsMap
+}
+
+func getEntriesMapOfComponentFromKeysList(component string, keys []string) map[string]bool  {
+	var entriesMap = make(map[string]bool)
+	for _,value := range keys {
+		value := strings.TrimPrefix(value, componentsPath)
+		parts := strings.Split(value, "/" )
+		if parts[0] == component {
+			entriesMap[parts[1]] = true
+		}
+	}
+	return entriesMap
+}
+
+func isFileExtensionValid(extension string) bool{
+	extension = strings.ToUpper(extension)
+	return extension == "JSON" || extension == "YAML" || extension == "INI" || extension == "TOML"
```

Comment:
Let's also allow `.yml` just in case. Visible strings should not be changed, just the logic to approve the file extension.